src/common/snippets/src/lowered/pass/iter_handler.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/lowered/pass/iter_handler.hpp"
+
+#include "snippets/itt.hpp"
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/lowered/loop_manager.hpp"
+#include "snippets/lowered/pass/propagate_subtensors.hpp"
+#include "snippets/snippets_isa.hpp"
+#include "snippets/utils.hpp"
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+UpdateMemoryAccessOps::UpdateMemoryAccessOps(size_t count) : RangedPass(), m_count(count) {}
+
+bool UpdateMemoryAccessOps::run(LinearIR& linear_ir, LinearIR::constExprIt begin, LinearIR::constExprIt end) {
+    for (auto expr_it = std::next(begin); expr_it != end; expr_it++) {
+        // Skip inner Loops
+        const auto loop_begin = ov::as_type_ptr<op::LoopBegin>(expr_it->get()->get_node());
+        if (loop_begin) {
+            expr_it = linear_ir.find(expr_it, end, linear_ir.get_expr_by_node(loop_begin->get_loop_end()));
```

Comment:
Thanks for the finding, the condition is added

Reply:
I'd expect to see exception here instead of just returning the status, what do you think?
Because the method updates MA in the Loop and we pass iterators of the Loop bounds (as it's mentioned in the brief description). If there is inner Loop, it must be finished in the outer Loop, not after

Reply:
Actually the pass realization doesn't require calling the pass on the Loop bounds range (maybe it is worth removing mention of Loop bounds from the description). There is nothing stopping us from reducing the range, at which the pass is be called, through e.g. some optimization. In this case, it may coincide that the optimized range turns out to be the inner Loop bounds. In my opinion, there is nothing criminal in such a scenario, and we should not throw an exception.

Reply:
Yeah, we should remove `Loop` mention from the description. Thanks~

Reply:
Already done